### DISCUSSION TITLE
Database reset when updating my schema

### DISCUSSION THREAD

**Discussion by rivaslive** [keystonejs/keystone#8953] [Questions] [answered]
Why when I make any changes to an existing model does the DB have to be emptied? I want to bring keystonejs to production but this worries me that tomorrow the production DB will disappear...

**Comment by borisno2** [comment, marked as answer]
Hey,
A destructive database migration will only happen when running `keystone dev` which is not recommended against a production database as it will reset the db if a clean migration cannot be performed. Have a look at the [database migration guide](https://keystonejs.com/docs/guides/database-migration) in the keystone docs for more information on database migrations.

**Reply by rivaslive** [reply]
Couldn't automatic migrations be incorporated? Some CMS have no problems with creating migrations and creating new or edited fields in the DB, I don't want to seem intense, I just would like to not have so many problems when modifying a field in a table in my DB

**Reply by dcousens** [reply]
@rivaslive did you read the migration guide? There is automatically generated migrations.
